react/utils/utils.ts

```diff
@@ -0,0 +1,15 @@
+export const validateInputs = (inputs: { [key: string]: any }): boolean => {
+  for (const key in inputs) {
+    if (inputs[key] === 'object') {
+      validateInputs(inputs[key])
+    }
+
+    const value = inputs[key]
+
+    if (!value) {
+      return false
+    }
+  }
+
+  return true
+}
```

Comment:
I think there is a mistake in this function in the recursive logic. It doesn't affect us because we are only checking `{ key: string }`, but if we try to apply it in a structure typed in the argument  `{ key: any }` it can give us a wrong result. [Here](https://replit.com/@FiladelfoBraz/AlienatedHumongousGui#index.js) one example showing that.

Also, if we are validating a number field, it will return false if value is zero, which might be a expected value.

Reply:
Good catch!